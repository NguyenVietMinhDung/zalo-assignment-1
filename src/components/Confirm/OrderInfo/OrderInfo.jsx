import React from 'react';
import { Container, Row, Col } from 'react-grid-system';
import './OrderInfo.scss';
import Product from '../../Categories/Category/Product';

const products = [
  {
    id: 0,
    name: 'Latte - This is a long text - Please use truncate the text if it is longer than 2 lines',
    description: 'Note - This text is not longer than 1 line',
    price: 99000,
    img: '270_crop_LATTE.png',
    quantity: 2,
    editable: true,
  },
  {
    id: 1,
    name: 'Cappuccino',
    description: 'Cappuccino',
    price: 39000,
    img: '270_crop_CAPPUCINO.png',
    quantity: 1,
    editable: true,
  },
];

const OrderInfo = () => (
  <div className="order-info">
    <Container className="order-info__container">
      <Row>
        <Col xs={12}>
          <h3>Thông tin đơn hàng</h3>
        </Col>
      </Row>

      {products.map(product => <Product key={product.id} {...product} />)}

      <Row className="order-info__cost">
        <Col xs={6}>
          Tạm tính
        </Col>
        <Col xs={6}>
          117.000đ
        </Col>
      </Row>

      <Row className="order-info__cost">
        <Col xs={6}>
          Mã giảm giá
        </Col>
        <Col className="order-info__cost__discount-value" xs={6}>
          -10.000đ
        </Col>
      </Row>

      <Row className="order-info__cost">
        <Col xs={6}>
          Tổng tiền
        </Col>
        <Col className="order-info__cost__total-value" xs={6}>
          107.000đ
        </Col>
      </Row>
    </Container>
  </div>
);

export default OrderInfo;
