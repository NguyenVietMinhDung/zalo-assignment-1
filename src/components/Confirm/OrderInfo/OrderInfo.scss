@import '../../../styles/variables';

.order-info {
	background-color: $white-smoke;

	&__container {
		display: flex;

		flex-direction: column;

		margin-bottom: 8px;

		background-color: white;
	}

	&__cost {
		div:last-child {
			text-align: right;
		}

		padding-top: 15px;
		padding-bottom: 15px;

		border-bottom: 1px solid $white-smoke;

		font-size: 12px;
		font-weight: 500;

		&__discount-value {
			color: green;
		}

		&__total-value {
			font-size: 16px;
			font-weight: 500;
		}
	}
}
