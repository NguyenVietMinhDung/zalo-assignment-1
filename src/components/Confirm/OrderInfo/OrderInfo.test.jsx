import React from 'react';
import { shallow } from 'enzyme';
import OrderInfo from './OrderInfo';

describe('<OrderInfo />', () => {
  test('renders correctly', () => {
    const component = shallow(<OrderInfo />);
    expect(component).toMatchSnapshot();
  });
});
