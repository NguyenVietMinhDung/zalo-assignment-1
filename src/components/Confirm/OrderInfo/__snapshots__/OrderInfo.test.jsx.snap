// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderInfo /> renders correctly 1`] = `
<div
  className="order-info"
>
  <Container
    className="order-info__container"
    component="div"
    fluid={false}
    lg={false}
    md={false}
    sm={false}
    style={Object {}}
    xl={false}
    xs={false}
    xxl={false}
  >
    <Row
      align="normal"
      component="div"
      debug={false}
      gutterWidth={null}
      justify="start"
      nogutter={false}
      nowrap={false}
      style={Object {}}
    >
      <Col
        component="div"
        debug={false}
        lg={null}
        md={null}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={null}
        style={Object {}}
        width={null}
        xl={null}
        xs={12}
        xxl={null}
      >
        <h3>
          Thông tin đơn hàng
        </h3>
      </Col>
    </Row>
    <Product
      description="Note - This text is not longer than 1 line"
      editable={true}
      id={0}
      img="270_crop_LATTE.png"
      key="0"
      name="Latte - This is a long text - Please use truncate the text if it is longer than 2 lines"
      price={99000}
      quantity={2}
    />
    <Product
      description="Cappuccino"
      editable={true}
      id={1}
      img="270_crop_CAPPUCINO.png"
      key="1"
      name="Cappuccino"
      price={39000}
      quantity={1}
    />
    <Row
      align="normal"
      className="order-info__cost"
      component="div"
      debug={false}
      gutterWidth={null}
      justify="start"
      nogutter={false}
      nowrap={false}
      style={Object {}}
    >
      <Col
        component="div"
        debug={false}
        lg={null}
        md={null}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={null}
        style={Object {}}
        width={null}
        xl={null}
        xs={6}
        xxl={null}
      >
        Tạm tính
      </Col>
      <Col
        component="div"
        debug={false}
        lg={null}
        md={null}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={null}
        style={Object {}}
        width={null}
        xl={null}
        xs={6}
        xxl={null}
      >
        117.000đ
      </Col>
    </Row>
    <Row
      align="normal"
      className="order-info__cost"
      component="div"
      debug={false}
      gutterWidth={null}
      justify="start"
      nogutter={false}
      nowrap={false}
      style={Object {}}
    >
      <Col
        component="div"
        debug={false}
        lg={null}
        md={null}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={null}
        style={Object {}}
        width={null}
        xl={null}
        xs={6}
        xxl={null}
      >
        Mã giảm giá
      </Col>
      <Col
        className="order-info__cost__discount-value"
        component="div"
        debug={false}
        lg={null}
        md={null}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={null}
        style={Object {}}
        width={null}
        xl={null}
        xs={6}
        xxl={null}
      >
        -10.000đ
      </Col>
    </Row>
    <Row
      align="normal"
      className="order-info__cost"
      component="div"
      debug={false}
      gutterWidth={null}
      justify="start"
      nogutter={false}
      nowrap={false}
      style={Object {}}
    >
      <Col
        component="div"
        debug={false}
        lg={null}
        md={null}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={null}
        style={Object {}}
        width={null}
        xl={null}
        xs={6}
        xxl={null}
      >
        Tổng tiền
      </Col>
      <Col
        className="order-info__cost__total-value"
        component="div"
        debug={false}
        lg={null}
        md={null}
        offset={Object {}}
        pull={Object {}}
        push={Object {}}
        sm={null}
        style={Object {}}
        width={null}
        xl={null}
        xs={6}
        xxl={null}
      >
        107.000đ
      </Col>
    </Row>
  </Container>
</div>
`;
